common/src/main/java/com/datastrato/gravitino/dto/util/DTOConverters.java

```diff
@@ -771,6 +772,56 @@ public Audit auditInfo() {
     };
   }
 
+  /**
+   * Converts a TopicDTO to a Topic.
+   *
+   * @param topicDTO The topic DTO to be converted.
+   * @return The topic.
+   */
+  public static Topic fromDTO(TopicDTO topicDTO) {
+    return new Topic() {
+      @Override
+      public String name() {
+        return topicDTO.name();
+      }
+
+      @Override
+      public String comment() {
+        return topicDTO.comment();
+      }
+
+      @Override
+      public Map<String, String> properties() {
+        return topicDTO.properties();
+      }
+
+      @Override
+      public Audit auditInfo() {
+        return topicDTO.auditInfo();
+      }
+
+      @Override
+      public int hashCode() {
+        return Objects.hash(name(), comment(), properties(), auditInfo());
+      }
```

Comment:
The topic equals function only needs to compare the topic name

Reply:
No. In order to make sure that some update operations are valid, we should not only compare the topic name